### ISSUE TITLE
`routing`: `lnd` cannot shutdown due to `MissionControl`

### ISSUE THREAD

**Issue by yyforyongyu** [lightningnetwork/lnd#8912]
Found in this [build](https://github.com/lightningnetwork/lnd/actions/runs/9889130981/job/27314526139?pr=8892),
```
itest error from [revoked_uncooperative_close_retribution_remote_hodl:Dave]: timeout waiting for process to exit
    harness.go:628: 
            Error Trace:    /home/runner/work/lnd/lnd/lntest/harness.go:628
                                        /home/runner/work/lnd/lnd/itest/lnd_revocation_test.go:676
                                        /home/runner/work/lnd/lnd/itest/lnd_revocation_test.go:701
            Error:          Received unexpected error:
                            restart node got error: timeout waiting for process to exit
            Test:           TestLightningNetworkDaemon/tranche04/79-of-156/neutrino/revoked_uncooperative_close_retribution_remote_hodl/SIMPLE_TAPROOT
            Messages:       failed to restart node Dave
    harness.go:440: finished test: revoked_uncooperative_close_retribution_remote_hodl, start height=450, end height=453, mined blocks=3
```

Turns out the mission control store cannot be stopped, as this line `Mission control store ticker stopped` isn't found in the logs,
```
2024-07-11 09:45:21.444 [DBG] CRTR: Stopping mission control store ticker
2024-07-11 09:45:21.445 [DBG] CRTR: Stored mission control results: 6 added, 0 deleted
2024-07-11 09:45:21.453 [DBG] BTCN: Finished batch, 1 unspent outpoints
...
2024-07-11 10:01:20.356 [DBG] HLCK: invoking success callback
```

Likely introduced from #8549, where `sync.Cond` was introduced,
https://github.com/lightningnetwork/lnd/blob/09b38aaa01d524afb1802d70d97d63183adcf334/routing/missioncontrol_store.go#L45-L46

### Analysis

Somehow the shutdown process was blocked. As indicated from the logs, after `stop` was called, the following lines were called, which drained the ticker,
1. https://github.com/lightningnetwork/lnd/blob/09b38aaa01d524afb1802d70d97d63183adcf334/routing/missioncontrol_store.go#L330
2. https://github.com/lightningnetwork/lnd/blob/09b38aaa01d524afb1802d70d97d63183adcf334/routing/missioncontrol_store.go#L335-L338 then blocks on the ticker.
Which is fixed in #8892.

Another potential issue is the usage of `sync.Cond`,
1. after calling `Signal` here, https://github.com/lightningnetwork/lnd/blob/09b38aaa01d524afb1802d70d97d63183adcf334/routing/missioncontrol_store.go#L285
2. We then wait here,
https://github.com/lightningnetwork/lnd/blob/09b38aaa01d524afb1802d70d97d63183adcf334/routing/missioncontrol_store.go#L313

I wonder if we could be blocked at the `Wait`. In hindsight, we should stick to the "communicate via go channels" pattern when it comes to concurrency management to make it easier to analyze issues as such.

**Comment by ellemouton** [comment]
@yyforyongyu  - is this solved now with that PR from last week?

**Comment by yyforyongyu** [comment]
Fixed by #9249
